napari/utils/key_bindings.py

```diff
@@ -182,6 +186,203 @@ def normalize_key_combo(key_combo):
+            return
+        for k, v in self._actions.items():
+            if function == v.command:
+                # print('this is an already existing command !')
+                self.bind_shortcut(k, shortcut)
+                return
+        # print('add', function, 'to orphans')
+        self._orphans[function] = shortcut
+
+    def _update_buttons_tooltips(self, name):
+        if name not in self._actions:
+            return
+        buttons = self._buttons[name]
+        desc = self._actions[name].description
+
+        if name in self._shortcuts:
+            sht = self._shortcuts[name]
+            desc += f' ({format_shortcut(sht)})'
+        for button, instances in buttons:
+            button.setToolTip(desc)
+
+    def _update_shortcut_bindings(self, name):
+        if name not in self._actions:
+            return
+        action = self._actions[name]
+        if name not in self._shortcuts:
+            return
+        sht = self._shortcuts.get(name)
+        keymappable = action.keymappable
+        if hasattr(keymappable, 'bind_key'):
+            keymappable.bind_key(sht)(action.command)
+
+    def bind_button(self, name, button, instance):
+        """
+        Bind `button` to trigger Action `name` on click.
+        """
+        (
+            command,
+            description,
+            _,
+            _icons,
+            _predicates,
+            keymappable,
+        ) = self._actions[name]
+        button.clicked.connect(lambda: command(instance))
+
+        self._buttons[name].append((button, instance))
+        self._update_buttons_tooltips(name)
+
+    def bind_shortcut(self, name, shortcut):
+        """
+        bind shortcut `shortcut` to trigger action `name`
+        """
+        self._shortcuts[name] = shortcut
+        self._update_buttons_tooltips(name)
+        self._update_shortcut_bindings(name)
+
+    def unbind_shortcut(self, name, shortcut):
+        action = self._actions[name]
+        sht = self._shortcuts.get(name)
+        keymappable = action.keymappable
+        if hasattr(keymappable, 'bind_key'):
+            keymappable.bind_key(sht)(None)
+        del self._shortcuts[name]
+        self._update_buttons_tooltips(name)
+
+
+action_manager = ActionManager()
+
+
+_actions = {
+    'V': 'toggle_selected_visibility',
+    'Control-G': 'toggle_grid',
+    'Shift-Down': 'also_select_layer_below',
+    'Shift-Up': 'also_select_layer_above',
+    'Down': 'select_layer_below',
+    'Up': 'select_layer_above',
+    'Control-A': 'select_all',
+    'Control-T': 'transpose_axes',
+    'Control-E': 'roll_axes',
+    'Alt-Down': 'focus_axes_down',
+    'Alt-Up': 'focus_axes_up',
+    'Right': 'increment_dims_right',
+    'Left': 'increment_dims_left',
+    'Control-Y': 'toggle_ndisplay',
+    'Control-Shift-Backspace': 'remove_all_layers',
+    'Control-Shift-Delete': 'remove_all_layers',
+}
+
+for (shortcut, action_name) in _actions.items():
+    action_manager.bind_shortcut(action_name, shortcut)
```

Comment:
@goanpeca and @ppwadhwa what would be the format (and the place) to load that in order to be the easiest from a configuration and internationalization perspective ?

How much do we want the possibility to unbind a shortcut instead of replacing it with another sequence ?